apollo-router/tests/common.rs

```diff
@@ -175,21 +173,80 @@ impl IntegrationTest {
-            .with_tracer(tracer)
-            .with_filter(LevelFilter::INFO);
-        let subscriber = Registry::default().with(telemetry).with(
-            tracing_subscriber::fmt::Layer::default()
-                .compact()
-                .with_filter(EnvFilter::from_default_env()),
-        );
+    fn init_telemetry(telemetry: Option<Telemetry>) {
+        match telemetry {
+            Some(Telemetry::Jaeger) => {
+                let tracer = opentelemetry_jaeger::new_agent_pipeline()
+                    .with_service_name("my_app")
+                    .install_simple()
+                    .expect("jaeger pipeline failed");
+                let telemetry = tracing_opentelemetry::layer()
+                    .with_tracer(tracer)
+                    .with_filter(LevelFilter::INFO);
+                let subscriber = Registry::default().with(telemetry).with(
+                    tracing_subscriber::fmt::Layer::default()
+                        .compact()
+                        .with_filter(EnvFilter::from_default_env()),
+                );
 
-        let _ = tracing::subscriber::set_global_default(subscriber);
-        global::set_text_map_propagator(propagator);
+                let _ = tracing::subscriber::set_global_default(subscriber);
+                global::set_text_map_propagator(opentelemetry_jaeger::Propagator::new());
+            }
+            Some(Telemetry::Datadog) => {
+                let tracer = opentelemetry_datadog::new_pipeline()
+                    .with_service_name("my_app")
+                    .install_simple()
+                    .expect("datadog pipeline failed");
+                let telemetry = tracing_opentelemetry::layer()
+                    .with_tracer(tracer)
+                    .with_filter(LevelFilter::INFO);
+                let subscriber = Registry::default().with(telemetry).with(
+                    tracing_subscriber::fmt::Layer::default()
+                        .compact()
+                        .with_filter(EnvFilter::from_default_env()),
+                );
+
+                let _ = tracing::subscriber::set_global_default(subscriber);
+                global::set_text_map_propagator(opentelemetry_datadog::DatadogPropagator::new());
+            }
+            Some(Telemetry::Otlp) => {
+                let tracer = opentelemetry_otlp::new_pipeline()
+                    .tracing()
+                    .install_simple()
+                    .expect("otlp pipeline failed");
+                let telemetry = tracing_opentelemetry::layer()
+                    .with_tracer(tracer)
+                    .with_filter(LevelFilter::INFO);
+                let subscriber = Registry::default().with(telemetry).with(
+                    tracing_subscriber::fmt::Layer::default()
+                        .compact()
+                        .with_filter(EnvFilter::from_default_env()),
+                );
+
+                let _ = tracing::subscriber::set_global_default(subscriber);
```

Comment:
Could the creation of the subscriber be extracted and made common? I think it could, but only if it's possible to set the propagator before the global default is set.

Reply:
The subscriber has to be boxed to do this. I did give this a quick go, wasn't sure that it added much, and the propagator cannot be boxed without creating a dyn wrapper. 

While not ideal, I think setting the subscriber and the propagator in consecutive statements is a nice thing to retain.